mitodl/mitxonline
# 3783: dedp course certificates in hubspot

### What are the relevant tickets?
https://github.com/mitodl/hq/issues/3783

### Description

1. Creates two new checkbox properties, associated with contacts, in Hubspot: course_run_certificates and program_certificates.
2. Adds functionality to populated the new properties with data based on an MITx Online learner's course run and program certificates.  Ensures that the HubSpot contact is updated whenever a certificate is created or deleted in MITx Online.

### How can this be tested?

1. Configure you local MITx Online instance to work with the Hubspot dev environment: (`MITOL_HUBSPOT_API_PRIVATE_TOKEN`, `MITOL_HUBSPOT_API_ID_PREFIX`, `HUBSPOT_PIPELINE_ID`).
3. Create a user and a corresponding course or program certificate.
4. Run `docker-compose run web ./manage.py configure_hubspot_properties` in order to update the Hubspot course_run_certificates and program_certificates properties with the string representation of your course runs and programs.
5.  From the profile page, update your user's first or last name in order to trigger a sync with Hubspot.
6. Verify that the corresponding Contact in Hubspot has the course or program certificate from step 2 defined.

The Hubspot Contact will have properties displayed like the screenshot below:
![Screenshot 2024-05-07 at 11 26 46 AM](https://github.com/mitodl/mitxonline/assets/8311573/363c9630-614a-449b-b6c4-2bc699b6733a)

You can also test the course and program certificate creation, and subsequent triggering of syncing the user's data with the Hubspot contact, by enrolling the user in a course, defining a grade for the user in edx, running the periodic edx sync command that checks for course run completion and grades from MITx Online, which will then create a course and possibly a program certificate.
